### PR TITLE
Fixing Applications retrieval related flow

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Purpose
Fixing several application retrieval flow issues.

### CODE REVIEW

File: common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApplicationDAOImpl.java [hunk 1]
```diff
@@ -140,6 +141,19 @@ private String getGroupId(Connection connection, int applicationId) throws SQLEx
...
     }
+    public Integer getSubscriptionCountByApplicationId(Application application, String organization)
+            throws APIManagementException {
+
+        int subscriptionCount = 0;
+
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+            return getSubscriptionCountByApplicationId(connection, application, organization);
+        } catch (SQLException e) {
+            handleException("Failed to get SubscribedAPI of : application " + application.getName() + " in " +
```
Comment (RakhithaRR) [inline common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApplicationDAOImpl.java:152]: Let's use exception codes for error handling. https://github.com/wso2/apk/blob/main/common-java-libs/org.wso2.apk.apimgt.api/src/main/java/org/wso2/apk/apimgt/api/ExceptionCodes.java

File: common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApplicationDAOImpl.java [hunk 2]
```diff
@@ -140,6 +141,19 @@ private String getGroupId(Connection connection, int applicationId) throws SQLEx
...
+                    "organization " + organization, e);
+        }
+        return subscriptionCount;
```
Comment (Krishanx92) [inline common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApplicationDAOImpl.java:155]: Do we need this? its always 0

Comment (CrowleyRajapakse) [reply common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApplicationDAOImpl.java:155]: this method calls for private method and returns the correct count, else will return 0

File: common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApplicationDAOImpl.java (hunk 1 above)
Comment (CrowleyRajapakse) [reply common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApplicationDAOImpl.java]: resolved

**Review by Krishanx92 — APPROVED** [review]

**Pull request merged**